src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -397,6 +706,21 @@ private BoundExpression ConvertConcatExprToString(SyntaxNode syntax, BoundExpres
                 }
             }
 
+            // If expression is of form `constantChar.ToString()` then rewrite it here to a string literal so we can choose better options for lowering later (e.g. fold it with another constant instead of performing concatenation)
+            // NOTE: We request `object.ToString()` as an optional member here to avoid reporting "missing member" diagnostic earlier than we should.
+            // Because if in the end it turns out that we have a simple concatenation of n strings there is no need to report missing `object.ToString()` member
+            if (TryGetSpecialTypeMethod(expr.Syntax, SpecialMember.System_Object__ToString, out MethodSymbol? optionalObjectToString, isOptional: true) &&
```

Comment:
>SpecialMember.System_Object__ToString [](http://example.com/codeflow?start=53&length=37)

Can we simply request the method from the compilation? We aren't really interested in any validation that ```TryGetSpecialTypeMethod``` is possibly going to perform, we just want to compare against the symbol if one is available.